### PR TITLE
ESP32: Add AM2320 support

### PR DESCRIPTION
**Pull request by thornley-david**
- [X] This PR is for the `dev-esp32` branch rather than for `master`.
- [X] This PR is compliant with the [other contributing guidelines](https://github.com/nodemcu/nodemcu-firmware/blob/dev/CONTRIBUTING.md) as well (if not, please describe why).
- [X] I have thoroughly tested my contribution.
- [X] The code changes are reflected in the documentation at `docs/*`.

Ported almost raw from ESP8266 implementation (https://github.com/nodemcu/nodemcu-firmware/blob/release/app/modules/am2320.c).

### CODE REVIEW

**Comment by marcelstoer** [comment]
I'd like to land #3135 from @nwf first and then for you to rebase.

**Comment by thornley-david** [comment]
No problems, this can definitely wait.

**Comment by nwf** [comment]
Giving the C a quick once-over, there doesn't seem to be any reason that this should be a C module (or ESP32 specific) and not just some Lua code: the sensor communicates over I2C, there appear to be no timing requirements, and all math is on small integers.  Maybe see #3197 or the similar Lua modules (e.g., bh1750 or ds2321 or mcp23008)?

**Comment by thornley-david** [comment]
@nwf, good point. I was just trying to reuse/migrate was was already there, however if there is a better way I am happy to rewrite it.

There are a couple of microsecond timing delays in the read negotiation, but they are likely not strict. To be honest, I am not aware if these can be transitioned to lua efficiently, and I was also not sure if the methodology I used was even correct when migrating (os_delay_us -> ets_delay_us). 

I'll take a look at the sample modules in detail and disband the PR if I managed to rewrite it as a lua module.

File: components/modules/Kconfig [hunk 1]
```diff
@@ -54,6 +54,12 @@ config LUA_MODULE_ADC
       Includes the adc module. This module provides access to the
       adc1 hardware.
 
+config LUA_MODULE_AM2320
```
Comment (marcelstoer) [inline components/modules/Kconfig:57]: In case you rebase this will have to become `NODEMCU_CMODULE_AM2320`.

**Pull request closed without merging**